### PR TITLE
get-childitem <PATH>/* -file should include <Path> as search directory

### LINKED ISSUE #5269: Strange behavior with Get-ChildItem
Steps to reproduce
------------------

```powershell
$Path='C:\temp'
#remove-item "$Path\TestPS"  -recurse
md "$Path\TestPS"
get-childitem -path "$Path\TestPS\F*.txt" -File -Recurse #OK no file

md "$Path\TestPS\D1"
get-childitem -path "$Path\TestPS" -Directory -Recurse #Ok one directory
get-childitem -path "$Path\TestPS\*" -Directory -Recurse #NOK no result

"Test" > "$Path\TestPS\D1\File1.txt"
get-childitem -path "$Path\TestPS\F*.txt" -File -Recurse #OK one file
get-childitem -path "$Path\TestPS" -File -Recurse #OK one file

get-childitem -path "$Path\TestPS\*" -File -Recurse #NOK no result

#Add a file into the 'root' of the search 
"Test" > "$Path\TestPS\File2.txt"
get-childitem -path "$Path\TestPS\*" -File -Recurse #OK two files !!! Different !!!
get-childitem -path "$Path\TestPS\*" -Directory -Recurse #OK one directory !!! Different !!! 


remove-item "$Path\TestPS\File2.txt"
#The 'root' of the search is empty 
get-childitem -path "$Path\TestPS\*" -File -Recurse #NOK no result
get-childitem -path "$Path\TestPS\*" -Directory -Recurse #NOK no result

#original path
#Get-ChildItem -Path 'C:\Users\laurent\AppData\Local\Mozilla\Firefox\Profiles\*.default*' -File -Rec

#it works
#Get-ChildItem -Path 'C:\Users\laurent\AppData\Local\Mozilla\Firefox\Profiles\*.default*' -Rec|Where-Object {-not $_.PSIsContainer}
```

Expected behavior
-----------------
The following calls must return existing files or directories :
```Powershell
get-childitem -path 'C:\Temp\TestPS\*" -File -Recurse
get-childitem -path 'C:\Temp\TestPS\*" -Directoyr -Recurse
```

Actual behavior
---------------
When the 'root' of the search ('C:\Temp\TestPS') is empty, the search do not return all entries :

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.0-beta.9
PSEdition                      Core
GitCommitId                    v6.0.0-beta.9
OS                             Microsoft Windows 6.1.7601 S
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0, 5.0, 5.1.10032.0, 6.0.0}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```
Same problem with For Windows PowerShell v 5.1.14409.1012 :
```powershell
> $PSVersionTable 
Name                           Value
----                           -----
PSVersion                      5.1.14409.1012
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0, 5.0, 5.1.14409.1012}
BuildVersion                   10.0.14409.1012
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### PR DESCRIPTION
**Pull request by chunqingchen**
This is to resolve #5269

When get-childitem is ended with * and -recurse, we only include the parent path as search path if the given path is a itemcontainer. We will return null if the parent directory only contains 1 sub folder with no sub folder under that folder. As if the parent folder only contains 1 subfolder, "*" will be mapped to that subfolder, and since it doesn't contain any subfolder, return is null.

However, the following scenario should also be included:
Given path is not a itemcontainer but -file parameter is specified 

$Path='C:\temp'
md "$Path\TestPS"
md "$Path\TestPS\D1"

"Test" > "$Path\TestPS\D1\File1.txt"
get-childitem -path "$Path\TestPS" -File -Recurse 

Before fix:
returns nothing

After fix:
File1.txt will be returned.

### CODE REVIEW

**Comment by iSazonov** [comment]
@chunqingchen Your commits contain updates for Pester - please fix (remove).

**Comment by chunqingchen** [comment]
@iSazonov your comment is resolved

**Comment by adityapatwardhan** [comment]
@anmenaga  Can you have a look? I have made some changes and pushed them to this PR.

**Comment by anmenaga** [comment]
@adityapatwardhan This code path is used heavily and is a high-risk area. I would like to see more tests, basically all cases that @LaurentDardenne mentions in #5269

**Comment by adityapatwardhan** [comment]
@anmenaga Can you have another look please?

File: src/System.Management.Automation/engine/SessionStateContainer.cs [hunk 1]
```diff
@@ -1509,9 +1525,8 @@ internal Collection<PSObject> GetChildItems(string[] paths, bool recurse, uint d
                             // Should glob paths and files that match tem*, but then
                             // recurse into all subdirectories and do the same for
                             // those directories.
-                            if ((!String.IsNullOrEmpty(path)) && (!IsItemContainer(path)))
+                            if ((!String.IsNullOrEmpty(path)) && ((!IsItemContainer(path))))
```
Comment (anmenaga) [inline src/System.Management.Automation/engine/SessionStateContainer.cs]: extra parenthesis?

**Review by anmenaga — APPROVED** [review]

**Pull request merged**